datacube/index/postgis/_datasets.py

```diff
@@ -161,19 +161,21 @@ def add(self, dataset: Dataset,
             return dataset
         with self._db_connection(transaction=True) as transaction:
             # 1a. insert (if not already exists)
-            transaction.insert_dataset(dataset.metadata_doc_without_lineage(), dataset.id, dataset.product.id)
```

Comment:
The `product_id` logic should be added in but the rest should be kept as it is here, not changed back

Reply:
As above.